src/scan-imports.ts

```diff
@@ -1,159 +1,131 @@
+import nodePath from 'path';
 import fs from 'fs';
-import chalk from 'chalk';
-import ora from 'ora';
 import glob from 'glob';
 import {parse} from '@babel/parser';
-import traverse, {NodePath} from '@babel/traverse';
-import {
-  CallExpression,
-  ImportDeclaration,
-  isImportDeclaration,
-  isCallExpression,
-} from '@babel/types';
+import traverse from '@babel/traverse';
 
 /**
- * Splits a filename into folder segments
+ * An install target represents information about a dependency to install.
+ * The specifier is the key pointing to the dependency, either as a package
+ * name or as an actual file path within node_modules. All other properties
+ * are metadata about what is actually being imported.
  */
-function explode(file: string) {
-  return file.split('/').filter(part => part);
+export type InstallTarget = {
+  specifier: string;
+  all: boolean;
+  default: boolean;
+  namespace: boolean;
+  named: string[];
+};
+
+function stripJsExtension(dep) {
+  return dep.replace(/\.js$/, '');
```

Comment:
nit: sometimes I like to throw a case-insensitive flag on RegExs just to be safe (`/\.js$/i/`)